packages/labs/react/src/use-controller.ts

```diff
@@ -131,23 +136,46 @@ export const useController = <C extends ReactiveController>(
     const host = new ReactControllerHost<C>(kickCount, kick);
     const controller = createController(host);
     host._primaryController = controller;
-    host._connected();
     return host;
   });
 
   host._updatePending = true;
 
-  // We use useLayoutEffect because we need updated() called synchronously
-  // after rendering.
-  useLayoutEffect(() => host._updated());
-
   // Returning a cleanup function simulates hostDisconnected timing. An empty
   // deps array tells React to only call this once: on mount with the cleanup
   // called on unmount.
-  useLayoutEffect(() => () => host._disconnected(), []);
+  //
+  // Note, it's important to call `connected` here and not when the host
+  // is created in the `useState` call since connection typically produces
+  // side effects and `useState` is expected to be pure.
+  // To preserve expected lifecycle ordering, the first update is also called
+  // here.
+  //
+  // This means that the initial value a controller produces may be incorrect
+  // since the React component sees this value before the controller has run
+  // connected+update.
+  //
+  // We could add an option to do a `requestUpdate` instead of an `update`
+  // here, but this would trigger a React re-render so it should only be done
+  // if required. Users can work around this initial value limitation by using
+  // the controller value in a `useLayoutEffect` and/or calling `requestUpdate`
+  // on the controller.
+  useLayoutEffect(() => {
+    host._connected();
+    host._update();
```

Comment:
I think it may be problematic that `_update()` is called after a render here.

Reply:
Updated with alternate gambit PTAL